streamline/base.py

```diff
@@ -136,6 +141,26 @@ class method.
         """
         return cls.name or cls.get_generic_name()
 
+    @staticmethod
```

Comment:
That has occurred to me, but, thing is these methods only ever update the props on the `RouteBase` class specifically, and not whatever class they are called from. So I figured I'd play safe and make them static. What do you think?